core/drivers/fabric-driver/server/listener.ts

```diff
@@ -0,0 +1,198 @@
+): Promise<void> => {
+    const listener: BlockListener = async (event) => {
+        // Parse the block data; typically there is only one element in this array but we will interate over it just to be safe
+        const blockData = ((event.blockData as fabproto6.common.Block).data as fabproto6.common.BlockData).data;
+        blockData.forEach((item) => {
+            const payload = Object.values(item)[Object.keys(item).indexOf('payload')];
+            const payloadData = Object.values(payload)[Object.keys(payload).indexOf('data')];
+            const transactions = payloadData.actions;
+            // Iterate through the transaction list
+            transactions.forEach(async (transaction: any) => {
+                // Get transaction chaincode ID
+                const chaincodeId = transaction.payload.chaincode_proposal_payload.input.chaincode_spec.chaincode_id.name;
+                if (transaction.payload.chaincode_proposal_payload.input.chaincode_spec.input.args.length > 0) {
+                    // Get transaction function name: first argument according to convention
+                    const chaincodeFunc = transaction.payload.chaincode_proposal_payload.input.chaincode_spec.input.args[0].toString();
+                    console.log('Trying to find match for channel', channelId, 'chaincode', chaincodeId, 'function', chaincodeFunc);
+                    // Find all matching event subscriptions stored in the database
+                    let eventMatcher = new events_pb.EventMatcher();
+                    eventMatcher.setEventType(events_pb.EventType.LEDGER_STATE);
+                    eventMatcher.setEventClassId('');       // Only match subscriptions where class ID is not specified
+                    eventMatcher.setTransactionLedgerId(channelId);
+                    eventMatcher.setTransactionContractId(chaincodeId);
+                    eventMatcher.setTransactionFunc(chaincodeFunc);
+                    let eventSubscriptionQueries;
+                    for (let i = 0 ; i < 10 ; i++) {
+                        try{
+                            eventSubscriptionQueries = await lookupEventSubscriptions(eventMatcher);
+                            break;
+                        } catch(error) {
+                            let errorString: string = error.toString();
+                            if (!errorString.includes('LEVEL_LOCKED')) {    // Check if contention was preventing DB access
+                                throw(error);
+                            }
+                            await new Promise(f => setTimeout(f, 2000));    // Sleep 2 seconds
+                        }
+                    }
+                    // Iterate through the view requests in the matching event subscriptions
+                    eventSubscriptionQueries.forEach(async (eventSubscriptionQuery: query_pb.Query) => {
+                        console.log('Generating view and collecting proof for channel', channelId, 'chaincode', chaincodeId);
```

Comment:
Do you want to add `transaction func` in the log as well?

Reply:
Done